libraries/rig-package/src/RigConfig.ts

```diff
@@ -199,80 +206,107 @@ export class RigConfig {
    * equal to `false`.
    */
   public static loadForProjectFolder(options: ILoadForProjectFolderOptions): RigConfig {
-    const rigConfigFilePath: string = path.join(options.projectFolderPath, 'config/rig.json');
+    const { overrideRigJsonObject, projectFolderPath } = options;
 
-    let json: IRigConfigJson;
-    try {
-      if (options.overrideRigJsonObject) {
-        json = options.overrideRigJsonObject;
-      } else {
-        if (!fs.existsSync(rigConfigFilePath)) {
-          return new RigConfig({
-            projectFolderPath: options.projectFolderPath,
-
-            rigFound: false,
-            filePath: '',
-            rigPackageName: '',
-            rigProfile: ''
-          });
-        }
+    const fromCache: RigConfig | false | undefined =
+      !options.bypassCache && !overrideRigJsonObject && RigConfig._configCache.get(projectFolderPath);
+    if (fromCache) {
+      return fromCache;
+    }
+
+    const rigConfigFilePath: string = path.join(projectFolderPath, 'config/rig.json');
 
+    let config: RigConfig | undefined;
+    let json: IRigConfigJson | undefined = overrideRigJsonObject;
+    try {
+      if (!json) {
         const rigConfigFileContent: string = fs.readFileSync(rigConfigFilePath).toString();
-        json = JSON.parse(stripJsonComments(rigConfigFileContent));
+        json = JSON.parse(stripJsonComments(rigConfigFileContent)) as IRigConfigJson;
       }
       RigConfig._validateSchema(json);
     } catch (error) {
-      throw new Error(error.message + '\nError loading config file: ' + rigConfigFilePath);
+      config = RigConfig._handleConfigError(error, projectFolderPath, rigConfigFilePath);
     }
 
-    return new RigConfig({
-      projectFolderPath: options.projectFolderPath,
+    if (!config) {
+      config = new RigConfig({
+        projectFolderPath: projectFolderPath,
 
-      rigFound: true,
-      filePath: rigConfigFilePath,
-      rigPackageName: json.rigPackageName,
-      rigProfile: json.rigProfile || 'default'
-    });
+        rigFound: true,
+        filePath: rigConfigFilePath,
+        rigPackageName: json!.rigPackageName,
+        rigProfile: json!.rigProfile || 'default'
```

Comment:
The constructor is private, so we can probably make `'default'` the default if this property is undefined.